homework/App.js

```diff
@@ -17,13 +17,27 @@ class App {
     // 2. Make an initial XMLHttpRequest using Util.fetchJSON() to populate your <select> element
 
     const root = document.getElementById('root');
-
-    Util.createAndAppend('h1', root, { text: 'It works!' }); // TODO: replace with your own code
+    const header = Util.createAndAppend('header', root, { class: 'header' });
+    Util.createAndAppend('p', header, { text: 'HYF Repositories' });
+    const selector = Util.createAndAppend('select', header, { class: 'repo-selector' });
+    const container = Util.createAndAppend('div', root, { id: 'container' });
+    const leftDiv = Util.createAndAppend('div', container, { class: 'left-div' });
 
     try {
       const repos = await Util.fetchJSON(url);
-      this.repos = repos.map(repo => new Repository(repo));
-      // TODO: add your own code here
+      this.repos = repos
+        .sort((a, b) => a.name.localeCompare(b.name))
+        .map(repo => new Repository(repo));
+      for (let i = 0; i < repos.length; i++) {
+        Util.createAndAppend('option', selector, { text: repos[i].name, value: i });
+      }
+      this.repos[0].render(leftDiv);
```

Comment:
The `.render()` method of the `Repository` object is also called in `fetchContributorsAndRender()`, line 73. This redundant call at initialization time should not be there.